### PR TITLE
Zoom plugin improvements

### PR DESCRIPTION
**Pull request by cmorbitzer**
This makes changes to `ZoomPlugin`, which was added in #3276. Each change is on its own commit on this branch, to make reversions/modifications easier if requested before approval.

### Fix scroll when using trackpad
The zoom is currently triggered by the wheel event. On a trackpad, horizontal scrolling fires a wheel event with both an X- and Y- delta, unless the user scrolls perfectly horizontally on the trackpad, which is not typical. This creates a poor UX when the user is attempting to scroll because the waveform errantly zooms simultaneously. This PR changes the event listener to perform a zoom only when the Y-delta is greater than the X-delta, which is the case for the pinch gesture.

### Account for wheel speed when zooming (breaking change)
Currently the zoom level is adjusted by the same amount for each wheel event that is fired. This PR changes the event listener to adjust the zoom level in proportion to the scroll/pinch distance. This offers a more consistent UX than relying on the quantity of event instances. As a result, `scale` changes from a magnification factor to a coefficient. A scale of `.2` felt too unresponsive with the new formula so this changes the default to `.5`.

### Allow using zoom plugin when `minPxPerSec` is 0 or default
The zoom formula that this change introduces swapped `scale` for `minPxPerSec` as the coefficient. Therefore, it is now possible to use this plugin without first having set `minPxPerSec` to a non-zero value.

 ### Add `maxZoom` option
Optional. Specifies a maximum `minPxPerSec` value while zooming.

## Checklist
* [ ] This PR is covered by e2e tests
* [ ] It introduces no breaking API changes

### CODE REVIEW

File: src/plugins/zoom.ts [hunk 1]
```diff
@@ -52,29 +53,28 @@ class ZoomPlugin extends BasePlugin<ZoomPluginEvents, ZoomPluginOptions> {
...
-    const pointerTime = (scrollX + x) / oldMinPxPerSec
-    const newMinPxPerSec = oldMinPxPerSec * (e.deltaY > 0 ? 1 - this.options.scale : 1 + this.options.scale)
-    const newLeftSec = (width / newMinPxPerSec) * (x / width)
+    const newMinPxPerSec = this.calculateNewZoom(oldMinPxPerSec, -e.deltaY)
     if (newMinPxPerSec * duration < width) {
       this.wavesurfer.zoom(width / duration)
-      this.container.scrollLeft = 0
     } else {
       this.wavesurfer.zoom(newMinPxPerSec)
-      this.container.scrollLeft = (pointerTime - newLeftSec) * newMinPxPerSec
```
[the remark this answers is not in the document]
Comment (cmorbitzer) [reply src/plugins/zoom.ts:74]: I'd actually like some confirmation on that. In v7.3, this code was necessary to accomplish that. I noticed that for me in v7.5.0, the scroll position stayed the same even when I removed this code. I don't know if that's true universally, or just for my environment or configuration.

[the remark this answers is not in the document]
Comment (katspaugh) [reply src/plugins/zoom.ts:74]: You probably know that but just to clarify:

The default zoom behavior is to preserve the scroll position at the playhead position. E.g. you click on a 5-second mark, playhead goes there, then you zoom and your scroll position remains on the 5-second mark.

With this plugin, the playhead position is ignored and it instead tries to stay at the same time position where you last scrolled.

So I'm wondering if this change alters this behavior.

[the remark this answers is not in the document]
Comment (cmorbitzer) [reply src/plugins/zoom.ts]: Based on the use of `e.clientX`, I actually think it is supposed to zoom in/out from the cursor position. So if the cursor is at the right side of the waveform, zooming in will create the appearance that the waveforms are stretching out to the left. If the cursor is in the middle, the stretch effect will occur evenly from both sides of the cursor. This is regardless of playhead and even scroll position. @HoodyHuo could confirm

If that's the case, this code is still relevant regardless of changes in the latest version. I added the code back in, just to be safe anyway.

**Comment by ffxsam** [comment]
Quick question:

I noticed (using a mouse wheel), that holding shift doesn't have the expected result of scrolling/panning through the waveform horizontally. Would you want to include that improvement in this PR, or open a new one?

**Comment by katspaugh** [comment]
@ffxsam that's a good idea! Please feel free to create a separate issue for it.

**Review by katspaugh — APPROVED** [review]

**Comment by katspaugh** [comment]
@ffxsam I've added your request in #3389. 👍 

@cmorbitzer thanks again for the amazing contribution! It's now released in 7.5.1. 🚀

**Comment by ffxsam** [comment]
@cmorbitzer @katspaugh I took this for a spin on my MacBook Air, and it works great! It's also pretty standard when compared against pro audio tools (Audition, iZotope RX, etc), in that down = zoom in, up = zoom out (or the opposite, depending on your Mac trackpad settings). 😆 

The scrolling _and_ panning action is perfect. Love this!

**Comment by ffxsam** [comment]
Hey guys, do you suppose there ought to be some sort of debouncing to save CPU cycles? This is pretty intense. 🤯 

https://github.com/katspaugh/wavesurfer.js/assets/12532733/e8daee7a-e4ff-4914-8377-97304258e592

**Comment by ffxsam** [comment]
Adobe Audition's zoom, for reference:

https://github.com/katspaugh/wavesurfer.js/assets/12532733/44682b21-2287-4138-bbb1-dc2d3c066afa

**Comment by katspaugh** [comment]
I suggest we continue this discussion [in the forum](https://github.com/katspaugh/wavesurfer.js/discussions/categories/general).

**Pull request merged**